rskj-core/src/test/java/co/rsk/vm/opcode/TransientStorageDslTest.java

```diff
@@ -45,23 +47,59 @@ void testTransientStorageOpcodesExecutionsWithRSKIPActivated() throws FileNotFou
+
+        String checkingOpcodesTxName2 = "txTestTransientStorageOpCodesOtherValue";
+        TransactionReceipt txReceipt2 = assertTransactionReceiptWithStatus(world, checkingOpcodesTxName2, "b04", true);
+        Assertions.assertEquals(1, TransactionReceiptUtil.getEventCount(txReceipt2, "OK", null));
+    }
+
+    @Test
+    void testTransientStorageOpcodesShareMemorySameTransaction() throws FileNotFoundException, DslProcessorException {
+        DslParser parser = DslParser.fromResource("dsl/transaction_storage_rskip446/tload_tstore_basic_contract.txt");
+        World world = new World();
+        WorldDslProcessor processor = new WorldDslProcessor(world);
+        processor.processCommands(parser);
+
+        String mainContractTransientStorageCreationTxName = "txTestTransientStorageContract";
+        assertTransactionReceiptWithStatus(world, mainContractTransientStorageCreationTxName, "b01", true);
+
+        String secondaryContractTransientStorageCreationTxName = "txTestTransientStorageOtherContract";
+        assertTransactionReceiptWithStatus(world, secondaryContractTransientStorageCreationTxName, "b02", true);
+
+        String checkingOpcodesTxName = "txTestTransientStorageNestedTransactionShareMemory";
+        TransactionReceipt txReceipt = assertTransactionReceiptWithStatus(world, checkingOpcodesTxName, "b05", true);
+        Assertions.assertEquals(1, TransactionReceiptUtil.getEventCount(txReceipt, "OK", null));
     }
 
     @Test
-    void testTransientStorageOpcodesExecutionsWithRSKIPDeactivated() throws FileNotFoundException, DslProcessorException {
+    void testTransientStorageOpcodesDoesntShareMemoryFromOtherContract() throws FileNotFoundException, DslProcessorException {
+        DslParser parser = DslParser.fromResource("dsl/transaction_storage_rskip446/tload_tstore_basic_contract.txt");
+        World world = new World();
+        WorldDslProcessor processor = new WorldDslProcessor(world);
+        processor.processCommands(parser);
+
+        String mainContractTransientStorageCreationTxName = "txTestTransientStorageContract";
+        assertTransactionReceiptWithStatus(world, mainContractTransientStorageCreationTxName, "b01", true);
+
+        String secondaryContractTransientStorageCreationTxName = "txTestTransientStorageOtherContract";
+        assertTransactionReceiptWithStatus(world, secondaryContractTransientStorageCreationTxName, "b02", true);
+
+        String checkingOpcodesTxName = "txTestTransientStorageNestedTransactionOtherContractDoesntShareMemory";
+        TransactionReceipt txReceipt = assertTransactionReceiptWithStatus(world, checkingOpcodesTxName, "b06", true);
+        Assertions.assertEquals(1, TransactionReceiptUtil.getEventCount(txReceipt, "ERROR", new String[]{"bytes32"}));
+    }
+
+    @Test
+    void testTransientStorageOpcodesExecutionFailsWithRSKIPDeactivated() throws FileNotFoundException, DslProcessorException {
```

Comment:
In the code the OP code call is done before the activation check, is the test working anyway?

Reply:
It is working, because it fails for the other one that was fine. We can have a split test with a contract that does only TLOAD and other only TSTORE to catch these two situations, I will do.